pss/outbox/outbox_test.go

```diff
@@ -0,0 +1,77 @@
+package outbox_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/ethersphere/swarm/pss/message"
+	"github.com/ethersphere/swarm/pss/outbox"
+)
+
+func TestOutbox(t *testing.T) {
+
+	outboxCapacity := 2
+	failed := make([]interface{}, 0)
+	failedC := make(chan struct{})
+	successC := make(chan struct{})
+	continueC := make(chan struct{})
+
+	forwardFail := false
+
+	mockForwardFunction := func(msg *message.Message) error {
+		if !forwardFail {
+			successC <- struct{}{}
+			return nil
+		} else {
+			failed = append(failed, msg)
+			failedC <- struct{}{}
+			<-continueC
+			return errors.New("forced test error forwarding message")
+		}
+	}
+
+	testOutbox := outbox.NewMock(&outbox.Config{
+		NumberSlots: outboxCapacity,
+		Forward:     mockForwardFunction,
+	})
+
+	testOutbox.Start()
+	defer testOutbox.Stop()
+
+	err := testOutbox.Enqueue(testOutboxMessage)
+	if err != nil {
+		t.Fatalf("unexpected error enqueueing, %v", err)
+	}
+
+	//We wait for the forward function to success
+	<-successC
+
+	forwardFail = true
+
+	err = testOutbox.Enqueue(testOutboxMessage)
+	if err != nil {
+		t.Fatalf("unexpected error enqueueing, %v", err)
+	}
+	//We wait for the forward function to fail
+	<-failedC
+
+	failedMessages := len(failed)
```

Comment:
Please consider whether this check is necessary.

Reply:
I'm removing this